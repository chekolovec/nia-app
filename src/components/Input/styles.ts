import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    padding: 20,
  },
  title: {
    fontSize: 25,
    paddingBottom: 20,
    textAlign: "center",
  },
  slider: {
    width: "80%",
  },
  sliderContainer: {
    flexDirection: "row",
    width: "100%",
    justifyContent: "space-between",
  },
  badgeContainer: {
    justifyContent: "center",
    alignItems: "center",
  },
  sliderWrapper: { flex: 1, alignItems: "center" },
});
