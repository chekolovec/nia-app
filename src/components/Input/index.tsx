import React from "react";
import Slider from "@react-native-community/slider";
import { Text, View } from "react-native";

import styles from "./styles";

interface IProps {
  title: string;
  onChange: (value: number) => void;
  value: number;
}

export const Input = ({ title, onChange, value }: IProps) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{title}</Text>
      <View style={styles.sliderContainer}>
        <View style={styles.badgeContainer}>
          <Text>0</Text>
          <Text>Bad</Text>
        </View>
        <View style={styles.sliderWrapper}>
          <Text testID="valueText">{value}</Text>
          <Slider
            style={styles.slider}
            minimumValue={0}
            maximumValue={10}
            step={1}
            onValueChange={onChange}
            value={value}
            minimumTrackTintColor="#FFFFFF"
            maximumTrackTintColor="#000000"
          />
        </View>
        <View style={styles.badgeContainer}>
          <Text>10</Text>
          <Text>Good</Text>
        </View>
      </View>
    </View>
  );
};
