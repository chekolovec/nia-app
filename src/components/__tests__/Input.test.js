import * as React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react-native";

import { Input } from "../Input";

it(`snapshots match`, () => {
  const tree = renderer
    .create(<Input title="Test Input title" onPress={() => {}} value={0} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("given a value Input should render it", () => {
  const mockFunc = jest.fn();
  const { getByTestId } = render(
    <Input title="Test Input title" onPress={mockFunc} value={5} />
  );
  const valueText = getByTestId("valueText");

  expect(valueText.props.children).toBe(5);
});
