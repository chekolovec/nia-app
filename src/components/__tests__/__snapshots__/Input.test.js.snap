// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots match 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 25,
        "paddingBottom": 20,
        "textAlign": "center",
      }
    }
  >
    Test Input title
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text>
        0
      </Text>
      <Text>
        Bad
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        testID="valueText"
      >
        0
      </Text>
      <RNCSlider
        disabled={false}
        enabled={true}
        inverted={false}
        maximumTrackTintColor="#000000"
        maximumValue={10}
        minimumTrackTintColor="#FFFFFF"
        minimumValue={0}
        onChange={null}
        onRNCSliderSlidingComplete={null}
        onRNCSliderSlidingStart={null}
        onRNCSliderValueChange={null}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        step={1}
        style={
          Array [
            Object {
              "height": 40,
            },
            Object {
              "width": "80%",
            },
          ]
        }
        thumbImage={
          Object {
            "__packager_asset": true,
            "height": 1,
            "scale": 1,
            "uri": "file://assets/full/path/to/directory/name.type",
            "width": 1,
          }
        }
        value={0}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text>
        10
      </Text>
      <Text>
        Good
      </Text>
    </View>
  </View>
</View>
`;
